src/core/renderers/GestureHandler.tsx

```diff
@@ -28,14 +29,36 @@ export const GestureHandler = ({
   notificationTopPosition,
 }: Props) => {
   const { width } = useWindowDimensions()
-  const notificationWidth = state.isPortaitMode
-    ? width - Constants.notificationSideMargin * 2
-    : Constants.maxNotificationWidth
+
+  const getDefaultWidth = () => width - Constants.notificationSideMargin * 2
+
+  const notificationWidthFromState = state?.notificationWidth
+  const hasNotificationWidth = !!notificationWidthFromState
+
+  const isWidthWithinBounds = hasNotificationWidth && notificationWidthFromState <= width
+
+  const getMaxWidth = (): number => {
+    if (hasNotificationWidth && notificationWidthFromState > width) {
+      return getDefaultWidth()
+    }
+    return hasNotificationWidth ? notificationWidthFromState : Constants.maxNotificationWidth
+  }
+
+  let notificationWidth: number
+  if (state.isPortrait) {
+    notificationWidth = isWidthWithinBounds ? notificationWidthFromState : getDefaultWidth()
+  } else {
+    notificationWidth = getMaxWidth()
+  }
```

Comment:
I think we can improve here a little bit: 

1. I would recommend moving these computations outside - it get's a little messy here, but if we create an utility function with 2-3 parameters, we can import it here, pass numbers, and assign the result to notificationWidth
2. I think this might be a little more readable:

```suggestion
  const fullWidth = width - Constants.notificationSideMargin * 2 // let's give it a better name, this is a full width of the screen with the margins

  const initialNotificationWidth = state?.notificationWidth || Constants.maxNotificationWidth; // let's name it initialNotificationWidht and assign to it user's value or default maxNotificationWidth - this way we simplify later logic

  const isWidthWithinBounds = notificationWidthFromState <= width

  const  notificationWidth = state.isPortrait && isWidthWithinBounds ? notificationWidthFromState : fullWidth // this is the shorthand from the if statement we had before (if (state.isPortrait) and then isWidthWithinBound are equal to state.IsPortrait && isWidthWithinBounds - any other situation we should return maxWidth
 
```

Just let me know what do you think about these little improvements, we can sync :)

By the way: good job!

Reply:
During testing, I noticed that notifications always display at full width when the device isn't in portrait mode. Since I removed the check for portrait mode, it's working correctly.